crates/rpc/rpc/src/eth/api/block.rs

```diff
@@ -11,9 +11,33 @@ where
 {
     pub(crate) async fn block_by_hash(
         &self,
-        _hash: H256,
+        hash: H256,
         _full: bool,
     ) -> EthResult<Option<RichBlock>> {
-        todo!()
+        let block = self.client().block(BlockId::Hash(hash.0.into()))?;
+        if let Some(_block) = block {
+            // TODO: GET TD FOR BLOCK - needs block provider? or header provider?
+            // let total_difficulty = todo!();
+            // let rich_block = Block::from_block_full(block, total_difficulty);
```

Comment:
not sure how this is stored but I think we need to add this to BlockProvider trait

Reply:
yeah, we'll need to add a method which obtains the total difficulty at a certain blocknum